offload/plugins-nextgen/common/src/GlobalHandler.cpp

```diff
@@ -238,7 +238,11 @@ void GPUProfGlobals::dump() const {
   for (const auto &ProfData : Data) {
     outs() << "{ ";
 #define INSTR_PROF_DATA(Type, LLVMType, Name, Initializer)                     \
-  outs() << ProfData.Name << " ";
+  if (sizeof(#Name) > 2 && #Name[sizeof(#Name) - 2] == ']') {                  \
```

Comment:
Thanks for catching this. Do you think you could add a comment explaining why you're making this check? Something like "Skip output for array fields".

Also, make sure to update the test. This change breaks offloading/pgo1.c because it doesn't expect the `[...]` output.

Reply:
Sure. Done.